File: src/main/scala/com/yammer/dropwizard/etc/CacheBustingFilter.scala
package com.yammer.dropwizard.etc

import javax.servlet._
import javax.servlet.http.HttpServletResponse

class CacheBustingFilter extends Filter {
  def init(filterConfig: FilterConfig) {}

  def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain) {
    if (response.isInstanceOf[HttpServletResponse]) {
      response.asInstanceOf[HttpServletResponse].setHeader("Cache-Control", "must-revalidate,no-cache,no-store")
    }
  }

  def destroy() {}
}

File: dropwizard-core/src/main/java/com/yammer/dropwizard/servlets/CacheBustingFilter.java
package com.yammer.dropwizard.servlets;

import javax.servlet.*;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class CacheBustingFilter implements Filter {
    @Override
    public void init(FilterConfig filterConfig) throws ServletException { /* unused */ }

    @Override
    public void doFilter(ServletRequest request,
                         ServletResponse response,
                         FilterChain chain) throws IOException, ServletException {
        if (response instanceof HttpServletResponse) {
            ((HttpServletResponse) response).setHeader("Cache-Control",
                                                       "must-revalidate,no-cache,no-store");
        }
        chain.doFilter(request, response);
    }

    @Override
    public void destroy() { /* unused */ }
}
